### PR TITLE
Change makefile allow specification of local/novel juju-gui branch.

### PR DESCRIPTION
**Pull request by bac**

### CODE REVIEW

**Comment by bac** [comment]
QA:

`make clean` - observe the directories `releases` and `jujugui-deps` do not exist.

`make package` - creates a `releases` and `jujugui-deps` directories. Ensure the `releases/jujugui-1.9.6.tar.bz2` exists and `jujugui-deps` is populated with lots of wheels.  Also note that in the output of the make command that juju-gui was pulled from github. You should see:

`git clone --depth 1 --branch master https://github.com/juju/juju-gui.git`

Now run `make releases` and observe it is a no-op.

Next ensure you can use a local branch for juju-gui, e.g.:

`JUJU_GUI_BRANCH=$HOME/git/juju-gui make package`

Again the tarball should be created but this time you'll git clone from the branch you specified.  You'll see something like the following in the make output:

`git clone --branch master /home/bac/git/juju-gui`

Also, test that `make clean unittest` creates the tarball and runs the tests a-ok.

**Comment by jaycee** [comment]
QAing this now.

**Comment by jaycee** [comment]
QA notes:

Attempting to do the step `JUJU_GUI_BRANCH= /home/jc/Code/canonical/juju-gui make package` I get:

```
find . -name '*.pyc' -delete
rm -rf tests/.venv
rm -rf tests/download-cache
rm -rf jujugui-deps
rm -rf releases
make -C src package
make[1]: Entering directory `/home/jc/Code/charms/trusty/juju-gui/src'
rm -rf juju-gui
git clone --branch master /home/jc/Code/canonical/juju-gui/
Cloning into 'juju-gui'...
fatal: Remote branch master not found in upstream origin
fatal: The remote end hung up unexpectedly
make[1]: *** [juju-gui] Error 128
make[1]: Leaving directory `/home/jc/Code/charms/trusty/juju-gui/src'
make: *** [releases] Error 2
```

**Comment by jaycee** [comment]
Once I checked out `master` into my repository, this worked; should this work with other branches, as the use case seems to me to be packaging local work?

File: Makefile [hunk 1]
```diff
@@ -102,4 +103,3 @@ help:
 
 .PHONY: all clean deploy ensure-juju-env ensure-juju-test ftest help \
     jujutest lint setup sysdeps test unittest
```
Comment (jaycee) [inline Makefile]: Not relevant for this branch, but we really need to move these .PHONY's to their corresponding targets.

**Comment by bac** [comment]
Good catch @jaycee.  Could you grab the new revision and try again?

File: src/Makefile [hunk 2]
```diff
@@ -14,10 +14,17 @@
...
+JUJU_GUI_BRANCH ?= 
 
+.PHONY: juju-gui
 juju-gui:
+	rm -rf juju-gui
+ifdef JUJU_GUI_BRANCH
+	git clone --branch master $(JUJU_GUI_BRANCH)
+else
 	git clone --depth 1 --branch master $(JUJU_GUI_PROJECT)
+endif
```
Comment (jaycee) [inline src/Makefile]: Nice!

File: src/Makefile [hunk 3]
```diff
@@ -14,10 +14,17 @@
...
+JUJU_GUI_BRANCH ?= 
 
+.PHONY: juju-gui
 juju-gui:
+	rm -rf juju-gui
+ifdef JUJU_GUI_BRANCH
+	git clone $(JUJU_GUI_BRANCH)
+else
 	git clone --depth 1 --branch master $(JUJU_GUI_PROJECT)
+endif
```
Comment (jaycee) [inline src/Makefile]: Nice!

**Comment by jaycee** [comment]
Code is :+1:; rechecking QA now.

**Comment by jaycee** [comment]
QA OK.

**Comment by hatched** [comment]
:+1: This looks awesome! Thanks for this @bac

**Comment by bac** [comment]
:shipit:

**Pull request merged**